### DISCUSSION TITLE
Rebuild widget every minute

### DISCUSSION THREAD

**Discussion by SasLuca** [rrousselGit/flutter_hooks#358] [Q&A]
I have a button which should be disabled after a certain hour every day.

In order to have it automatically change when necessary I thought of using a timer to rebuild the button every minute, so that all the checks get recomputed with the current `DateTime.now()`.

My current solution is this:

```dart
final now = useState(DateTime.now());

useEffect(()
{
  final timer = Timer.periodic(const Duration(minutes: 1), (_) => now.value = DateTime.now());
  return timer.cancel;
}, []);
```

I however don't really need that `now` variable, so I was curious if there is a way to get rid of it and just have the timer tell the widget to rebuild.

I am also open to other solutions if there are any.

**Comment by rrousselGit** [comment]
You don't need to make it a DateTime. You could make an integer instead and do `now.value++`

**Reply by rrousselGit** [reply]
You likely should also change that state into a "should button be enabled" instead of the current time

**Reply by SasLuca** [reply]
I am aware I dont need to make a DateTime, initially I thought of making use of that DateTime but then I realize I dont need but it stuck.

Is there however a way to just not have to use any state variable here and just tell the widget to rebuild, or is using a useless integer the best option?
